src/validation/validation.cpp

```diff
@@ -2915,6 +2915,15 @@ bool DisconnectTip(CValidationState &state, const Consensus::Params &consensusPa
     if (!FlushStateToDisk(state, FLUSH_STATE_IF_NEEDED))
         return false;
 
+    // If this block enabled the Nov 15th 2019 protocol upgrade, then we need to clear the mempool of any transaction
+    // using
+    // not previously avaiable features (e.g. Schnorr enabled OP_CHECKMULTISIG)
+    if (IsNov2019Enabled(consensusParams, pindexDelete) && !IsNov2019Enabled(consensusParams, pindexDelete->pprev))
+    {
+        mempool.clear();
```

Comment:
Is there a mempool admittance queue that also needs to cleared? ping @gandrewstone

Reply:
good question? I think it all depends on the fact if the txs in the queue are supposed to be already validate or not. Agree that @gandrewstone is the right person to ask to.

Reply:
yes there's a commitQ.   You can do this to clear it.

+boost::unique_lock<boost::mutex> lock(csCommitQ);
+txCommitQ.clear();

Reply:
on the other hand if 1902 gets merged then there's no need to clear the mempool here.

Reply:
@ptschip agreed. Will add the bit about `txCommitQ()` and will remove it later on  in case #1902 will get merged before this one.